src/entities.rs

```diff
@@ -224,6 +224,40 @@ impl Entities {
         }
     }
 
+    /// Allocate a specific entity ID and generation.  May return the location of the entity
+    /// previously allocated to the specified ID.
+    ///
+    /// Location should be written immediately.
+    pub fn alloc_at(&mut self, entity: Entity) -> Option<Location> {
+        self.verify_flushed();
+
+        let loc = if entity.id as usize >= self.meta.len() {
+            self.pending.extend((self.meta.len() as u32)..entity.id);
+            self.meta.resize(entity.id as usize + 1, EntityMeta::EMPTY);
+            None
+        } else if let Some(index) = self
+            .pending
+            .iter()
+            .find(|item| **item == entity.id)
+            .copied()
+        {
+            self.pending.swap_remove(index as usize);
```

Comment:
`index` is an element of `pending`, not a location within it. `swap_remove` takes a location within the vec.

Reply:
Adjusted condition to use `.position` so swap_remove should work correctly here.